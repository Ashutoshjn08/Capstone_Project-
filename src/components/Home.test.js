import { shallow } from "enzyme";
import Home from "./Home";
import toJson from "enzyme-to-json";
import renderer from "react-test-renderer";

test("renders correctly (snapshot)", () => {
  const wrapperSnap = renderer.create(<Home />);
  let tree = wrapperSnap.toJSON();
  expect(tree).toMatchSnapshot();
  console.log(toJson(wrapperSnap));
});

describe("renders correnctly", () => {
  let wrapper;

  beforeEach(()=>{
    wrapper = shallow(<Home/>)
  })
  
  test("heading renders correctly", ()=>{
    let wrapperHeading = wrapper.find(`[data-test='heading']`);
    expect(wrapperHeading.length).toBe(1)
  })
  test("text renders correctly", ()=>{
    let wrapperHeading = wrapper.find('p');
    expect(wrapperHeading.text()).toEqual('This system has provision to view the list of products and allows only the registered and authenticated users to add, edit, delete the products and view the product detail. The user interface to the system is implemented using React. The data about the products is managed in a JSON server. The client application interfaces with the JSON Server through http to retrieve and store data.')
  })
  test("text renders correctly", ()=>{
    let wrapperHeading = wrapper.find('span');
    expect(wrapperHeading.text()).toEqual('View Our Products')
  })
});
