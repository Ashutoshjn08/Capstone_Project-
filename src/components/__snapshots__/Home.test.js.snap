// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly (snapshot) 1`] = `
<div
  className="ui container"
>
  <div
    className="home-container"
  >
    <div
      className="card text-center"
    >
      <div
        className="card-header card-head"
        data-test="heading"
      >
        <h3>
          Product Inventory
        </h3>
      </div>
      <div
        className="card-body"
      >
        <p
          className="card-text"
          style={
            Object {
              "textAlign": "justify",
            }
          }
        >
          This system has provision to view the list of products and allows only the registered and authenticated users to add, edit, delete the products and view the product detail. The user interface to the system is implemented using React. The data about the products is managed in a JSON server. The client application interfaces with the JSON Server through http to retrieve and store data.
        </p>
        <a
          className="btn btn-primary"
          href="/products"
          style={
            Object {
              "width": "200px",
            }
          }
        >
          <span>
            View Our Products
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
